### ISSUE TITLE
Date changed programmatically is not updated

### ISSUE THREAD

**Issue by Prabhakar-Poudel** [edgycircle/ember-pikaday#158]
I have a modal that shows start and end of events. The start/end time can be altered using the datepicker.
However, if there are bunch of events for different dates, and user opens the modal by clicking at them the dates are not shown correctly.

Suppose, if we have following events:

```json
[
  {
    "name": "e1",
    "start": "2017-05-15T10:34:47+05:30",
    "end":  "2017-05-15T10:34:47+05:30"
  },
  {
    "name": "e2",
    "start": "2017-05-17T10:34:47+05:30",
    "end":  "2017-05-17T10:34:47+05:30"
  },
  {
    "name": "e3",
    "start": "2017-05-12T10:34:47+05:30",
    "end":  "2017-05-12T10:34:47+05:30"
  },
  {
    "name": "e4",
    "start": "2017-05-13T10:34:47+05:30",
    "end": "2017-05-13T10:34:47+05:30"
  }
]
```
Note: dates are converted to the required form before updating the `value` attributes of datepickers.

So, if I start with event `e1` and check the details, it shows correctly(note: only the first render works correctly).
Then if I check details of event `e4`, the start date shows correctly but the end date caches from the old result. (i.e start: 2017/05/13, end: 2017/05/15 - end value for `e1`)
Next, I do the same with `e3` it shows (start: 2017/05/12, end: 2017/05/13 - end value for `e4`)
Next if I repeat the same for event `e2` result is (start 2017/05/12 - start value for `e3`, end: 2017/05/17).

Notice how on decreasing dates, the start times shows the correct value but end time shows that of the previous event, details of which should have been updated and not even known to the datepicker. Exactly reverse of it happens when moving to increasing dates (i.e. end dates come up correctly while the start dates come from previous settings).

Another strange thing is I'm resetting both dates to null when the modal that shows it closes. So every time the modal shows up the value passes to the datepickers is updated.
I have checked up my code, tested if modal is messing it up, and all other possible things. Everything works fine except the dates are not rendered correctly.

My guess is on pikaday's `didUpdateAttrs` hook, there is `run.later` called to update values. And for some reason it is not getting triggered as expected.

It was a complex problem to provide twidle with. So I have made my best effort to explain the problem.

**Comment by Prabhakar-Poudel** [comment]
I found the source of this problem. And fix for it.
My `minDate` and `maxDate` were also programmaticlly updated, just before the modal showed up.
The problem was it was picking the previous vlues of `min` and `max` causing it to force dates to max and min of those set previously. Also, making the illusion of rendering previous values.
I tried putting null values for min and max when I'm done with the inputs, and it did not work. 
After going through ember/pikaday code I found it would execute `setMaxDate` and `setMinDate` only if a value is passed.
So, a simple fix was to set it to some value that is acceptable and beyond your actual value. Anyway the `min` and `max` would update eventually.
